x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/result_settings_field_number.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, FocusEvent } from 'react';
+
+import { EuiFieldNumber } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { SIZE_FIELD_MAXIMUM, SIZE_FIELD_MINIMUM } from '../constants';
+import { FieldResultSettingObject } from '../types';
+
+const updateOrClearSizeForField = (
+  fieldName: string,
+  fieldValue: number,
+  updateAction: (fieldName: string, size: number) => void,
+  clearAction: (fieldName: string) => void
+) => {
+  if (typeof fieldValue === 'number' && !isNaN(fieldValue)) {
+    updateAction(fieldName, fieldValue);
+  } else {
+    clearAction(fieldName);
+  }
+};
+
+const handleFieldNumberChange = (
+  fieldName: string,
+  updateAction: (fieldName: string, size: number) => void,
+  clearAction: (fieldName: string) => void
+) => {
+  return (e: ChangeEvent<HTMLInputElement>) => {
+    const fieldValue = parseInt(e.target.value, 10);
+    updateOrClearSizeForField(fieldName, fieldValue, updateAction, clearAction);
+  };
+};
+
+const handleFieldNumberBlur = (
+  fieldName: string,
+  updateAction: (fieldName: string, size: number) => void,
+  clearAction: (fieldName: string) => void
+) => {
+  return (e: FocusEvent<HTMLInputElement>) => {
+    const value = parseInt(e.target.value, 10);
+    const fieldValue = Math.min(
+      SIZE_FIELD_MAXIMUM,
+      Math.max(SIZE_FIELD_MINIMUM, isNaN(value) ? 0 : value)
+    );
+    updateOrClearSizeForField(fieldName, fieldValue, updateAction, clearAction);
+  };
+};
+
+interface Props {
+  fieldSettings: FieldResultSettingObject;
+  fieldName: string;
+  fieldEnabledProperty: string;
+  fieldSizeProperty: string;
+  updateAction: (fieldName: string, size: number) => void;
+  clearAction: (fieldName: string) => void;
+}
+
+export const ResultSettingsFieldNumber: React.FC<Props> = ({
```

Comment:
I pulled this into its own component, because the majority of the table's complexity comes from these fields.

It took me a little while to figure out why we had this onBlur, but then I realized that it's to avoid validation on this field. We prevent people from entering invalid value by correcting them on blur.

There were some weird cases that weren't covered that I cover here. Like for instance, in ent-search, you could enter "E" and it wouldn't be corrected.